src/bp/nlu-core/initialize-tools.ts

```diff
@@ -57,13 +59,32 @@ const initializeLanguageProvider = async (
   }
 }
 
-const initDucklingExtractor = async (config: NLU.LanguageConfig, logger: NLU.Logger): Promise<void> => {
-  await DucklingEntityExtractor.configure(config.ducklingEnabled, config.ducklingURL, logger)
+const makeSystemEntityExtractor = async (
+  config: NLU.LanguageConfig,
+  logger: NLU.Logger
+): Promise<SystemEntityExtractor> => {
+  let extractor: SystemEntityExtractor
+  if (config.ducklingEnabled) {
+    const duckCache = new SystemEntityCacheManager(
+      path.join(process.APP_DATA_PATH || '', 'cache', 'duckling_sys_entities.json'),
+      true,
+      logger
+    )
+    extractor = new DucklingEntityExtractor(duckCache, logger)
+    await extractor.configure(config.ducklingEnabled, config.ducklingURL)
+  } else {
+    const msCache = new SystemEntityCacheManager(
+      path.join(process.APP_DATA_PATH || '', 'cache', 'microsoft_sys_entities.json'),
```

Comment:
I see this `process.APP_DATA_PATH || ''` in a couple of places... If we had the same error, it occurred while running tests. I made a small change in another PR so we can avoid this `|| ''`, which is more a hack than anything

![image](https://user-images.githubusercontent.com/42552874/109082542-363b5580-76d2-11eb-8f6d-9934b3ac81e5.png)

Reply:
I don't get what you mean, my tests and code were working fine (just copied the old code but kept the path logic).  Do you suggest that I assign `' '` to process.APP_DATA_PATH in `initialize-tools.ts` ? Or that I override it in my tests ?